bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/core/IBulkMutation.java

```diff
@@ -15,38 +15,34 @@
  */
 package com.google.cloud.bigtable.core;
 
-import com.google.api.core.ApiFuture;
-import com.google.bigtable.v2.MutateRowRequest;
+import com.google.bigtable.v2.MutateRowResponse;
 import com.google.cloud.bigtable.data.v2.models.RowMutation;
 
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
+import com.google.common.util.concurrent.ListenableFuture;
 
 /**
- * Interface to support batching multiple RowMutation request in to singe grpc request.
+ * Interface to support batching multiple {@link RowMutation} request in to singe grpc request.
  */
 public interface IBulkMutation {
-  long MAX_RPC_WAIT_TIME_NANOS = TimeUnit.MINUTES.toNanos(12);
 
   /**
-   * Send any outstanding {@link MutateRowRequest}s and wait until all requests are complete.
+   * Send any outstanding {@link RowMutation} and wait until all requests are complete.
    */
-  void flush() throws InterruptedException, TimeoutException;
+  void flush() throws InterruptedException;
 
   void sendUnsent();
 
   /**
-   * @return false if there are any outstanding {@link MutateRowRequest} that still need to be sent.
+   * @return false if there are any outstanding {@link RowMutation} that still need to be sent.
    */
   boolean isFlushed();
 
   /**
-   * Adds a {@link com.google.cloud.bigtable.data.v2.models.RowMutation} to the underlying IBulkMutation
-   * mechanism.
+   * Adds a {@link RowMutation} to the underlying IBulkMutation mechanism.
    *
-   * @param rowMutation The {@link com.google.cloud.bigtable.data.v2.models.RowMutation} to add
-   * @return a {@link ApiFuture} of type {@link Void} will be set when request is
+   * @param rowMutation The {@link RowMutation} to add.
+   * @return a {@link ListenableFuture} of type {@link MutateRowResponse} will be set when request is
    *     successful otherwise exception will be thrown.
    */
-  ApiFuture<Void> add(RowMutation rowMutation);
+  ListenableFuture<MutateRowResponse> add(RowMutation rowMutation);
```

Comment:
I have changed to ListenableFuture in this interface because here in [BatchExecutor](https://github.com/rahulKQL/cloud-bigtable-client/blob/777e2ff9ff1f52e0a29454a2c6f69c6464af2f25/bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BatchExecutor.java#L201) we are calling multiple operations and all of them are resulting in Listenable.

I have checked in GCJ's version of [v2.models.BulkMutation](https://github.com/googleapis/google-cloud-java/blob/675ace7601e2c09474ce986d5d2223af5cb15f28/google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/BulkMutation.java#L80) it returns Bulktation only.

Reply:
++ @ajaaym